src/core.php

```diff
@@ -901,7 +1004,7 @@ protected function &__xhpChildrenDeclaration(): mixed {
   abstract protected function stringify(): string;
 
   final public function toString(): string {
-    $this->__flushElementChildren();
+    $this->__flushElementChildren()->join();
```

Comment:
Oh, I get what you're saying. You want to be able to do `$xhp->asyncToString();`